src/fork_gc.c

```diff
@@ -737,55 +847,61 @@ static bool FGC_parentHandleNumeric(ForkGC *gc, int *ret_val, RedisModuleCtx *rc
       FGC_unlock(gc, rctx);
       hasLock = 0;
     }
-    if (shouldReturn) {
-      if (fieldName) {
-        rm_free(fieldName);
-      }
-      return false;
-    }
+    rm_free(ninfo.valuesDeleted);
   }
 
-  if (fieldName) {
-    rm_free(fieldName);
-  }
-  return true;
+  rm_free(fieldName);
+  return status;
 }
 
-static bool FGC_parentHandleTags(ForkGC *gc, int *ret_val, RedisModuleCtx *rctx) {
+static FGCError FGC_parentHandleTags(ForkGC *gc, RedisModuleCtx *rctx) {
   int hasLock = 0;
   size_t fieldNameLen;
-  char *fieldName = FGC_recvBuffer(gc, &fieldNameLen);
-  if (fieldName == RECV_BUFFER_EMPTY) {
-    return false;
-  }
-
-  uint64_t tagUniqueId = FGC_recvLongLong(gc);
+  char *fieldName;
+  uint64_t tagUniqueId;
   InvertedIndex *value = NULL;
+  FGCError status = recvNumericTagHeader(gc, &fieldName, &fieldNameLen, &tagUniqueId);
```

Comment:
I would call it FGCStatus

Reply:
we already have FGCState, so Error is more distinguishing